go/fn/object_test.go

```diff
@@ -237,6 +238,38 @@ func TestSetNestedFields(t *testing.T) {
 	}
 }
 
+func TestInternalAnnotationsUntouchable(t *testing.T) {
+	o := NewEmptyKubeObject()
+	// Verify the "upstream-identifier" annotation cannot be changed via SetNestedStringMap
+	o.SetNestedStringMap(map[string]string{"owner": "kpt"}, "metadata", "annotations")
```

Comment:
Why not using the helper? https://pkg.go.dev/github.com/GoogleContainerTools/kpt-functions-sdk/go/fn#KubeObject.SetAnnotation
Same comment below.

Reply:
oh, this group of tests is to make sure there's no way users can change the upstream-identifier annotation via SDK methods. So we test all cases including setAnnotation.